engine/sound/src/decoders/decoder_opus.cpp

```diff
@@ -0,0 +1,502 @@
+
+                uint8_t segment_size = stream_info->m_lacing_table[stream_info->m_lacing_tableIndex];
+
+                if (segment_size > 0) {
+                    EnsureDataRead(stream_info, sound_data, STREAM_BLOCK_SIZE);
+
+                    if (stream_info->m_DataBuffer.Size() < segment_size) {
+                        break;
+                    }
+
+                    if (max_out_bytes < (stream_info->m_PacketOffset + segment_size))
+                    {
+                        assert(!"Larger packet buffer needed!");
+                        break;
+                    }
+
+                    memcpy(out_buffer + stream_info->m_PacketOffset, stream_info->m_DataBuffer.Begin(), segment_size);
+
+                    CleanupBuffer(stream_info->m_DataBuffer, segment_size);
+                    stream_info->m_PacketOffset += segment_size;
+                }
+
+                if (++stream_info->m_lacing_tableIndex == stream_info->m_lacing_tableSize) {
+                    stream_info->m_lacing_tableIndex = 256;
+                }
+
+                if (segment_size < 255) {
+                    out_bytes = stream_info->m_PacketOffset;
+                    stream_info->m_PacketOffset = 0;
+                    is_ok = true;
+                    break;
+                }
+            }
+            else {
+                // Skip however many bytes we need to skip...
+                while(stream_info->m_SkipBytes) {
+                    if (stream_info->m_DataBuffer.Empty()) {
+                        uint32_t read_size;
+                        dmSound::Result res = dmSound::SoundDataRead(sound_data, stream_info->m_StreamOffset, STREAM_BLOCK_SIZE, stream_info->m_DataBuffer.Begin(), &read_size);
+                        if (res == dmSound::RESULT_OK || res == dmSound::RESULT_PARTIAL_DATA) {
+                            stream_info->m_DataBuffer.SetSize(read_size);
+                            stream_info->m_StreamOffset += read_size;
+                        }
+                        else {
+                            stream_info->m_bEOS = (res == dmSound::RESULT_END_OF_STREAM);
+                            break;
+                        }
+                    }
+                    uint32_t chunk = dmMath::Min(stream_info->m_DataBuffer.Size(), stream_info->m_SkipBytes);
+                    CleanupBuffer(stream_info->m_DataBuffer, chunk);
+                    stream_info->m_SkipBytes -= chunk;
+                }
+            }
+        }
+        while(!(stream_info->m_bEOS && stream_info->m_DataBuffer.Empty()));
+        return is_ok;
+    }
+    
+    static Result OpusOpenStream(dmSound::HSoundData sound_data, HDecodeStream* stream)
+    {
+        //
+        // Note: the code below assumes all data needed for intialization of the stream is
+        //       already available and does NOT need to be stalled for!
+        //
+        Decodestream_info *stream_info = new Decodestream_info;
```

Comment:
`malloc` instead of `new`.

Reply:
As long as it's consistent, I don't mind new (it's still a .cpp file after all)